zerver/tests/test_users.py

```diff
@@ -1399,6 +1399,39 @@ def test_api(self) -> None:
         user = self.example_user("hamlet")
         self.assertTrue(user.is_active)
 
+    def test_email_sent(self) -> None:
+        self.login("iago")
+        user = self.example_user("hamlet")
+
+        with mock.patch("zerver.views.users.send_email") as m:
+            result = self.client_delete(
+                f"/json/users/{user.id}",
+                dict(
+                    deactivation_notification_comment="Dear Hamlet,\nyou just got deactivated.",
+                ),
+            )
+
+        self.assert_json_success(result)
+        self.assertTrue(m.called)
+        m.assert_called_with(
+            "zerver/emails/deactivate",
+            to_user_ids=[10],
+            from_address="noreply@testserver",
+            context={
+                "deactivation_notification_comment": "Dear Hamlet,\nyou just got deactivated.",
+                "realm_uri": "http://zulip.testserver",
+                "realm_name": "Zulip Dev",
+            },
+        )
+
+        do_reactivate_user(user, acting_user=None)
+        self.assertTrue(user.is_active)
+        with mock.patch("zerver.views.users.send_email") as n:
+            result = self.client_delete(f"/json/users/{user.id}", dict())
+
+        self.assert_json_success(result)
+        self.assertFalse(n.called)
```

Comment:
I rewrote this test to avoid mocking, and instead use the `outbox` to verify what actually happened -- check it out :)